models/commit_status.go

```diff
@@ -291,3 +284,32 @@ func ParseCommitsWithStatus(oldCommits *list.List, repo *Repository) *list.List
 	}
 	return newCommits
 }
+
+// HashCommitStatusContext hash context
+func HashCommitStatusContext(context string) string {
+	return fmt.Sprintf("%x", md5.Sum([]byte(context)))
+}
+
+// CommitStatusContext represents commit status context
+type CommitStatusContext struct {
+	ID          int64
+	RepoID      int64          `xorm:"index"`
+	ContextHash string         `xorm:"varchar(40) unique"`
```

Comment:
`unique` differs from model definition in models package.

Reply:
What do you mean here?

Reply:
You are using `unique` here and at the CommitStatusContext struct in the models package you are using `index` instead.

Reply:
Sorry nevermind. Got something wrong.

Reply:
Why is the contexthash unique? What happens if a context with the same text got reported?

Reply:
Before insert a new context, have checked if the context exist via checking the contexthash

Reply:
But what if a context got reported for multiple repos?